tests/testsuite.src/run_misc.at

```diff
@@ -4252,13 +4252,125 @@ AT_DATA([caller.cob], [
+       01 COUNTER PIC 9(5) VALUE 0.
+       
+       PROCEDURE DIVISION.
+           PERFORM 1000 TIMES
+              ADD 1 TO COUNTER
+              CALL "sched_yield"
+           END-PERFORM
+           DISPLAY "Final count: " COUNTER
+           EXIT PROGRAM.
+
+])
+AT_DATA([caller.c], [
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <pthread.h> /* For POSIX threads */
+#include <dlfcn.h>   /* For dynamic library loading */
+#include <unistd.h> 
+
+typedef int (*hello_world_func)(void);
+typedef void (*cob_init_func)(int, char **);
+typedef hello_world_func (*cob_resolve_func)(char *);
+typedef void (*cob_tidy_func)(void);
+/* Function to call COBOL shared library functions */
+void *callCobol() {
+    const char *libCobPath = "libcob.so";
+    void *libCobHandle;
+    cob_init_func cob_init;
+    cob_resolve_func cob_resolve;
+    cob_tidy_func cob_tidy;
+    hello_world_func hello_world;
+    
+    /* Use `dlmopen` to load a unique copy of the shared library in each thread */
+    libCobHandle = dlmopen(LM_ID_NEWLM, libCobPath, RTLD_NOW);
+    if (!libCobHandle) {
+        fprintf(stderr, "Thread %ld: Failed to load libcob with dlmopen from %s: %s\n",
+                pthread_self(), libCobPath, dlerror());
+        return NULL;
+    }
+
+#define RESOLVE_SYMBOL(symbol, type) \
+    symbol = (type) dlsym(libCobHandle, #symbol); \
+    if (!symbol) { \
+        fprintf(stderr, "Thread %ld: Failed to load symbol %s: %s\n", \
+                pthread_self(), #symbol, dlerror()); \
+        dlclose(libCobHandle); \
+        return NULL; \
+    }
+
+    /* Resolve required symbols from the COBOL library */
+    RESOLVE_SYMBOL(cob_init, cob_init_func);
+    RESOLVE_SYMBOL(cob_resolve, cob_resolve_func);
+    RESOLVE_SYMBOL(cob_tidy, cob_tidy_func);
+    cob_init(0, NULL);
+    hello_world = cob_resolve("prog");
+    if (hello_world) {
+        hello_world(); /* Call the COBOL function */
+    } else {
+        fprintf(stderr, "Thread %ld: Failed to resolve prog function.\n", pthread_self());
+    }
+    /* Clean up COBOL resources */
+    cob_tidy();
+    dlclose(libCobHandle);
+    return NULL;
+}
+/* Thread function wrapper around `callCobol` */
+void *threadFunction(void *arg) {
+    long threadId = (long)arg;
+    callCobol();
+    return NULL;
+}
+/* Main function */
+int main() {
+    const int numThreads = 2; /* Number of threads to create */
+    pthread_t threads[[numThreads]];
+    long i;
+    int j;
+    /* Create multiple threads */
+    for (i = 0; i < numThreads; i++) {
+        if (pthread_create(&threads[[i]], NULL, threadFunction, (void *)i) != 0) {
+            fprintf(stderr, "Main thread: Failed to create thread %ld.\n", i);
+            return EXIT_FAILURE;
+        }
+    }
+    /* Wait for all threads to finish */
+    for (j = 0; j < numThreads; j++) {
+        pthread_join(threads[[j]], NULL);
+    }
+    return EXIT_SUCCESS;
+}
+])
+AT_CHECK([$COBC -x -ldl -lpthread caller.c], [0], [], [])
```

Comment:
You may skip the test if that compile fails, this reduces the necessary changes.
Something along:
```
AT_CHECK([$COBC -x -ldl -lpthread caller.c], [0], [], [],
    echo "C compilation failed, skipping rest of group"
    exit 77
)
```
(77 is the standard exit code for "skip")

... or change configure + aclocal as you did, but then also change build_windows/config.h.in